internal/bgp/bgp.go

```diff
@@ -519,8 +522,19 @@ type tcpmd5sig struct {
 // proper TCP MD5 options when the password is not empty. Works by manupulating
 // the low level FD's, skipping the net.Conn API as it has not hooks to set
 // the neccessary sockopts for TCP MD5.
-func dialMD5(ctx context.Context, addr, password string) (net.Conn, error) {
-	laddr, err := net.ResolveTCPAddr("tcp", "[::]:0")
+func dialMD5(ctx context.Context, addr string, srcAddr string, password string) (net.Conn, error) {
+	// If srcAddr exists on any of the local network interfaces, use it as the
+	// source address of the TCP socket. Otherwise, fall back to the IPv6
+	// unspecified address ("[::]") to let the kernel figure out the source
+	// address.
```

Comment:
This seems confusing, as "[::]" will of course use an IPv4 address if that is okay too. The text seems confusing in that regard.

I'd also say that this will set the src-address only if the field is set in the configuration (is optional)

Reply:
I've used the term "IPv6 unspecified address" because that's the official name of `::`. Anyway, I added a note which makes it clear that `::` also includes IPv4 on Linux.

Reply:
>I'd also say that this will set the src-address only if the field is set in the configuration (is optional)

Done.